pkg/config/config.go

```diff
@@ -124,47 +126,68 @@ func (c *Config) ValidateAndCompleteConfig() error {
 		}
 	}
 
-	if err := c.validatePythonPackagesHaveVersions(); err != nil {
-		return err
+	if len(c.Build.PythonPackages) > 0 && c.Build.PythonRequirements != "" {
+		return fmt.Errorf("Only one of python_packages or python_requirements can be set in your cog.yaml, not both")
 	}
 
-	if c.Build.GPU {
-		if err := c.validateAndCompleteCUDA(); err != nil {
+	// Load python_requirements into memory to simplify reading it multiple times
+	if c.Build.PythonRequirements != "" {
+		b, err := os.ReadFile(path.Join(projectDir, c.Build.PythonRequirements))
+		if err != nil {
 			return err
 		}
+		c.Build.pythonRequirementsContent = strings.Split(string(b), "\n")
 	}
 
-	if len(c.Build.PythonPackages) > 0 && c.Build.PythonRequirements != "" {
-		return fmt.Errorf("Only one of python_packages or python_requirements can be set in your cog.yaml, not both")
+	// Backwards compatibility
+	if len(c.Build.PythonPackages) > 0 {
+		c.Build.pythonRequirementsContent = c.Build.PythonPackages
+	}
+
+	if c.Build.GPU {
+		if err := c.validateAndCompleteCUDA(); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
-func (c *Config) PythonPackagesForArch(goos string, goarch string) (packages []string, indexURLs []string, err error) {
-	packages = []string{}
+// PythonRequirementsForArch returns a requirements.txt file with all the GPU packages resolved for given OS and architecture.
+func (c *Config) PythonRequirementsForArch(goos string, goarch string) (string, error) {
+	packages := []string{}
 	indexURLSet := map[string]bool{}
-	for _, pkg := range c.Build.PythonPackages {
+	for _, pkg := range c.Build.pythonRequirementsContent {
 		archPkg, indexURL, err := c.pythonPackageForArch(pkg, goos, goarch)
 		if err != nil {
-			return nil, nil, err
+			return "", err
 		}
 		packages = append(packages, archPkg)
 		if indexURL != "" {
 			indexURLSet[indexURL] = true
 		}
 	}
-	indexURLs = []string{}
+
+	// Create final requirements.txt output
+	// Put index URLs first
+	lines := []string{}
 	for indexURL := range indexURLSet {
-		indexURLs = append(indexURLs, indexURL)
+		lines = append(lines, "--find-links "+indexURL)
```

Comment:
Do we want to take the opportunity to clean up the nomenclature confusion here (`--index-url` is a different thing to `--find-links`) or is that more trouble than it's worth?

Reply:
We need `--index-url` for Torch 1.12/1.13 I think, so let's maybe clean up in a follow-up PR for that.